studio.py

```diff
@@ -14,14 +22,211 @@
+                )
+                if self.information["device_name"] == "Pixel 2":
+                    platform_name = "pixel2"
+                elif self.information["device_name"] == "Moto G (5)":
+                    platform_name = "motog5"
+
+            for recording in recordings_json:
+                label = recording.get("label", None)
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                        ]
+                    )
+                recording["path"] = os.path.join(self.path, name, name)
+                recording["name"] = name
+
+                recording["recording_path"] = "%s.mp" % recording["path"]
+                recording["json_path"] = "%s.json" % recording["path"]
+                recording["screen_shot_path"] = "%s.png" % recording["path"]
+                recording["zip_path"] = os.path.join(
+                    self.path, "%s.zip" % recording["name"]
+                )
+                recording["manifest_path"] = os.path.join(
+                    self.path, "%s.manifest" % recording["name"]
+                )
+
+                recordings.append(recording)
+        else:
+            raise Exception("No site file found!!!")
+
+        return recordings
+
+    def update_json_information(self, recording):
+        time.sleep(3)
+        print("Updating json with recording information")
+
+        with open(recording["json_path"], "r") as f:
+            json_data = json.loads(f.read())
+
+        self.information["proxy"] = self.proxy
+
+        self.information["url"] = recording["url"]
+        self.information["domain"] = recording["domain"]
+
+        label = recording.get("label", None)
+        if label:
+            self.information["label"] = label
+
+        json_data["info"] = self.information
+        with open(recording["json_path"], "w") as f:
+            f.write(json.dumps(json_data))
+
+    def generate_zip_file(self, recording):
+        print("Generating zip file")
+        zf = zipfile.ZipFile(recording["zip_path"], "w")
+        zf.write(
+            recording["recording_path"], os.path.basename(recording["recording_path"])
+        )
+        zf.write(recording["json_path"], os.path.basename(recording["json_path"]))
+        zf.write(
+            recording["screen_shot_path"],
+            os.path.basename(recording["screen_shot_path"]),
+        )
+        zf.close()
```

Comment:
ZipFile can be used as a context manager, so you don't need to worry about closing the file.

```python
with ZipFile('spam.zip', 'w') as myzip:
    myzip.write('eggs.txt')
```